### ISSUE TITLE
Wordlist thoughts/progress

### ISSUE THREAD

**Issue by SamCoVT** [SamCoVT/TaliForth2#2]
I'm currently working on my feature/wordlists branch.

I've reclaimed tmp3 from use in `FIND-NAME`.  It was being used to traverse the string in the dictionary (along with tmp2 being used to traverse the string the user provided).  I re-used tmp1 for this and pushed-pulled it as needed, so tmp3 is now available to traverse the wordlist order.

The current plan of attack is to put some words in another wordlist (such as the EDITOR wordlist), set the wordlists and search order up in the USER area of RAM in `COLD`, and then start writing `SEARCH-WORDLIST`, which will be reusing most of the code from `FIND-NAME`.  I'll leave `FIND-NAME` alone so Tali still works, and I can test by calling `SEARCH-WORDLIST` directly.

Once `SEARCH-WORDLIST` is complete, I plan to rewrite `FIND-NAME` using `SEARCH-WORDLIST` on each of the wordlists in the search order.  This will be the big switchover where everything will need to work.

**Comment by SamCoVT** [comment]
The big switchover is mostly complete, and I'm back to the point where all of the current tests pass.  I only have `MARKER` (which needs to save/restore the wordlists and the search order), and the words that need to be taught how to traverse the search order (`FIND`, `FIND-NAME`, `WORDS`, `INT>NAME`).  These currently only search the CURRENT (compilation) wordlist as they still use DP directly, but they pass all of the tests right now because the system starts off with FORTH-WORDLIST set up as CURRENT.

**Comment by SamCoVT** [comment]
Switchover is complete.  Now I just need to bring in the Search-Order tests from ANS and then update the documentation.  As a side note, I made `INT>NAME` search **all** of the wordlists regardless of the search order.  The other words that search the dictionary now honor the search order.

Yay for a testing suite.  It prevented me from really breaking things, or at least notified me when I just recently broke things.

**Comment by SamCoVT** [comment]
Wordlists are now working, with some basic documentation and the ANS tests!  The only word that initially failed the ANS tests was FIND, and it actually had a small bug (not that anyone noticed... who uses find these days?).  I did have to modify a couple of the tests that used `C"` because Tali doesn't have that (it makes an old-style counted string).  I just used `S\"` with the `\x` option to stuff the length in as the first byte, and that worked.  I'm glad I got around to writing `S\"`, as it's been super-useful recently for both of us.

I've implemented all of the search-order words, including the extended words.  The extended words are in forth_words.fs as the standard provided sample code for them, but some of them are simple enough I'll at least think about rewriting them in assembly before issuing the pull request.

The startup delay has now gotten long enough that I want to go back and assemblerize more of the words in forth_words.fs.  Perhaps once I'm done with the search order stuff, I'll revisit the block words.  Now that `SRC` and `BLK` are in the USER variables, I can rewrite most of that in assembly.  I can also put the editor into the EDITOR-WORDLIST, now that such a thing exists.

Just FYI, the EDITOR-WORDLIST and ASSEMBLER-WORDLIST both have just a single dummy word in them that I could use for testing.  They should be replaced with actual editor and assembler words at some point in the future.  I was thinking that ed and the block editor would live in the EDITOR-WORDLIST, unless you want to leave ed in the FORTH-WORDLIST.

**Comment by SamCoVT** [comment]
I wanted to start putting the simple screen editor I wrote into the `EDITOR-WORDLIST` in assembly, but then I realized that I would need the block words in assembly first, so that's what I'm currently working on.  I've moved the buffer variables into the user variables, and I figured out how to implement the deferred words `BLOCK-READ` and `BLOCK-WRITE`.  It took me a while to figure out that those words needed to have the CF (has CFA) flag in the dictionary (you get bonus points for your overcommenting in CREATE), and then a little while more to realize that they should never be natively compiled and add the NN flag.  I'm going to see how far through the block words I can get this weekend.  I'm looking forward to a reduction in startup time.
